src/commands/account/view_account_summary/mod.rs

```diff
@@ -124,32 +124,59 @@ fn get_account_inquiry(
     let runtime = tokio::runtime::Builder::new_multi_thread()
         .enable_all()
         .build()?;
-    let concurrency = 10;
+    // Staring from Feb 2025, the rate limit is 150 requests per 30 seconds for mainnet.
+    // We will limit the number of requests per batch to 140 to be conservative.
+    let batch_size = 140;
+    let batch_cooldown = tokio::time::Duration::from_secs(30);
+    let concurrency = 10; // Process 10 requests concurrently within each batch
+
     let delegated_stake: color_eyre::Result<
         std::collections::BTreeMap<near_primitives::types::AccountId, near_token::NearToken>,
     > = match validators {
```

Comment:
suggest renaming `validators`. @akorchyn  you mentioned it takes (it might take) ~10 batches on mainnet to iterate over. With the named batches size (140) this number is too big to represent validators. If it's not `validators` , it should be renamed accordingly